### PR TITLE
initial devops work, including requirements, style, and workflows

### PR DESCRIPTION
**Pull request by maconard**
- Implement github workflows for linting and testing pull requests
- Add a `LICENSE`
- Add a `pyproject.toml` to define the `tilewe` package
- Add a `setup.py` for use with packaging `tilewe`
- Add a `requirements.txt` file for use with `pip`, to install dependencies
- Add `flake8` style enforcement
  - Apply fixes for new linting rules across the code base
  - Specifically exclude rules: W291,W293,W504,E128,E201,E202,E252,E302,E305
- Add `scripts` folder, so far with a `validate_style.sh` script in it
- Add requirement for Python 3.10 (specifically python 3 and minor version >=10)
- Add pre-commit hooks for checking style and unit tests
- Add `tilewe.elo` unit tests

### CODE REVIEW

**Comment by maconard** [comment]
@nhamil now ready for review!

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
```
Comment (nhamil) [inline LICENSE:2]:
```suggestion
```

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+Copyright (c) 2023 Nicholas Hamilton, Michael Conard
```
Comment (nhamil) [inline LICENSE:3]:
```suggestion
Copyright (c) 2023 Nicholas Hamilton, Michael Conard
```

File: tilewe/__init__.py [hunk 3]
```diff
@@ -1,7 +1,11 @@
 from collections import defaultdict
+import sys
 
 import numpy as np 
 
+if sys.version_info[0] != 3 or sys.version_info[1] < 10:
+    raise Exception("Requires Python 3.10")
```
Comment (nhamil) [inline tilewe/__init__.py:7]:
```suggestion
    raise Exception("Requires Python 3.10+")
```

File: tilewe/engine.py [hunk 4]
```diff
@@ -114,10 +113,10 @@ def on_search(self, board: tilewe.Board, _seconds: float) -> tilewe.Move:
...
         random.shuffle(moves) 
 
-        best = max(moves, key=lambda m: \
-                        tilewe.n_piece_tiles(m.piece) * 100 + \
-                        tilewe.n_piece_corners(m.piece) * 10 + \
-                        tilewe.n_piece_contacts(m.piece))
+        best = max(moves, key=lambda m:
+            tilewe.n_piece_tiles(m.piece) * 100 +
+            tilewe.n_piece_corners(m.piece) * 10 +
+            tilewe.n_piece_contacts(m.piece))
```
Comment (nhamil) [inline tilewe/engine.py:119]: lol oops, good catch

File: tilewe/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+[project]
+name = "tilewe"
+version = "0.0.1"
+authors = [
+  { name="Nicholas Hamilton", email="nh.contact.1@gmail.com" },
+  { name="Michael Conard", email="michaelaconard314@gmail.com" },
+]
+description = "A tile placing game for AI development fun"
```
Comment (nhamil) [inline tilewe/pyproject.toml:8]:
```suggestion
description = "A tile-placing game for AI development fun"
```

File: tilewe/tests/test_elo.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        # 94.68% win rate against 500 weaker enemy
+        self.assertAlmostEqual(tilewe.elo.elo_win_probability(2000, 1500, C), 0.9468, 4)
+
+        #  5.32% win rate against 500 stronger enemy
+        self.assertAlmostEqual(tilewe.elo.elo_win_probability(1500, 2000, C), 0.0532, 4)
+        
+        # 99.01% win rate against 800 weaker enemy
+        self.assertAlmostEqual(tilewe.elo.elo_win_probability(2300, 1500, C), 0.9901, 4)
+
+        #  0.99% win rate against 800 stronger enemy
```
Comment (nhamil) [inline tilewe/tests/test_elo.py:29]:
```suggestion
        # 0.99% win rate against 800 stronger enemy
```

**Review by nhamil — APPROVED** [review]

**Pull request merged**